src/leaflet.groupedlayercontrol.js

```diff
@@ -66,7 +69,7 @@ L.Control.GroupedLayers = L.Control.extend({
     var id = L.Util.stamp(layer);
     var _layer = this._getLayer(id);
     if (_layer) {
-      delete this.layers[this.layers.indexOf(_layer)];
+      this._layers.splice(this._layers.indexOf(_layer), 1);
```

Comment:
This is a bug fix independent from my collapsable groups feature.

Reply:
Nice catch! Someone else opened a PR for this which I just merged (https://github.com/ismyrnow/leaflet-groupedlayercontrol/pull/50). Can you take this commit out of your branch?

Reply:
The fix from #50 is incorrect, as they use the delete keyword to remove an item from the _layers array, this will result in the array looking like [{object}, {object}, undefined, {object}] and when iterating through it elsewhere within your code, it will break when it expects to find an object and instead gets undefined.

The splice method should be used instead.